src/test/munit/tcp-test-suite.xml

```diff
@@ -0,0 +1,218 @@
+    </flow>
+
+    <flow name="tcp-xml-listen">
+        <sockets:listener config-ref="tcp-listener-xml-protocol"/>
+        <munit-tools:store key='message'/>
+    </flow>
+
+    <flow name="tcp-xml-eof-listen">
+        <sockets:listener config-ref="tcp-listener-xml-eof-protocol"/>
+        <munit-tools:store key='message'/>
+    </flow>
+
+    <flow name="tcp-safe-listen">
+        <sockets:listener config-ref="tcp-listener-safe-protocol"/>
+        <logger level="INFO" message="YYYYYYY"/>
+        <munit-tools:store key='message' >
+            <munit-tools:value>'OK'</munit-tools:value>
+        </munit-tools:store>
+    </flow>
+
+    <munit:test name="tcp-test" timeOut="220000">
+        <munit:enable-flow-sources>
+            <munit:enable-flow-source value="tcp-listen"/>
+        </munit:enable-flow-sources>
+        <munit:execution>
+            <set-payload value=''/>
+            <sockets:send-and-receive config-ref="tcp-requester">
+                <sockets:content><![CDATA['Hello world!']]></sockets:content>
+            </sockets:send-and-receive>
+        </munit:execution>
+        <munit:validation>
+            <until-successful maxRetries="60" millisBetweenRetries="2000">
+                <munit-tools:retrieve key='message'/>
+                <munit-tools:assert-that
+                        expression="#[payload.^raw as String]"
+                        is="#[MunitTools::equalTo('OK')]"
+                        message="Message not received"/>
+            </until-successful>
+        </munit:validation>
+    </munit:test>
+
+    <munit:test name="tcp-safe-test">
+        <munit:enable-flow-sources>
+            <munit:enable-flow-source value="tcp-safe-listen"/>
+        </munit:enable-flow-sources>
+        <munit:execution>
+            <set-payload value=''/>
+            <sockets:send-and-receive config-ref="tcp-requester-safe-protocol">
+                <sockets:content><![CDATA['Hello world!']]></sockets:content>
+            </sockets:send-and-receive>
+        </munit:execution>
+        <munit:validation>
+            <until-successful maxRetries="60" millisBetweenRetries="5000">
+                <munit-tools:retrieve key='message'/>
+                <munit-tools:assert-that
+                        expression="#[payload.^raw as String]"
+                        is="#[MunitTools::containsString('OK')]"
+                        message="Message not received"/>
+            </until-successful>
+        </munit:validation>
+    </munit:test>
+
+    <!-- incorrect err is returned  expected:SOCKETS:CONNECTI[ON_TIMEOUT] but was:SOCKETS:CONNECTI[VITY -->
+        <munit:test name="tcp-timeout-test" expectedErrorType="SOCKETS:CONNECTIVITY">
+            <munit:execution>
+                <set-payload value="Hello world!"/>
+                <sockets:send config-ref="tcp-requester-conn-timeout"/>
+            </munit:execution>
+            <munit:validation>
+                <logger message="#[payload]" level="INFO"/>
```

Comment:
Same here, are you expecting both to happen?

Reply:
same